scripts/Connector/Connector_Molex/conn_molex_micro-fit-3.0_tht_top_dual_row.py

```diff
@@ -0,0 +1,275 @@
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with kicad-footprint-generator. If not, see < http://www.gnu.org/licenses/ >.
+'''
+
+import sys
+import os
+#sys.path.append(os.path.join(sys.path[0],"..","..","kicad_mod")) # load kicad_mod path
+
+# export PYTHONPATH="${PYTHONPATH}<path to kicad-footprint-generator directory>"
+sys.path.append(os.path.join(sys.path[0], "..", "..", ".."))  # load parent path of KicadModTree
+from math import sqrt
+import argparse
+import yaml
+from helpers import *
+from KicadModTree import *
+
+sys.path.append(os.path.join(sys.path[0], "..", "..", "tools"))  # load parent path of tools
+from footprint_text_fields import addTextFields
+
+series = "Micro-Fit_3.0"
+series_long = 'Micro-Fit 3.0 Connector System'
+manufacturer = 'Molex'
+orientation = 'V'
+number_of_rows = 2
+
+variant_params = {
+    'solder_mounting':{
+        'mount_pins': 'solder', # remove this
+        'datasheet': 'http://www.molex.com/pdm_docs/sd/430450212_sd.pdf',
+        'C_minus_B': 6,
+        'part_code': "43045-{n:02}12",
+        'alternative_codes': [
+            "43045-{n:02}13",
+            "43045-{n:02}24"
+            ]
+        },
+}
+
+pins_per_row_range = range(1,13)
+pitch = 3.0
+drill = 1.0
+pad_size = [1.6, 1.6]
+peg_drill = 1.0
+
+def generate_one_footprint(pins_per_row, variant, configuration):
+    mpn = variant_params[variant]['part_code'].format(n=pins_per_row*2)
+    alt_mpn = [code.format(n=pins_per_row*2) for code in variant_params[variant]['alternative_codes']]
+
+    # handle arguments
+    orientation_str = configuration['orientation_options'][orientation]
+    footprint_name = configuration['fp_name_format_string'].format(man=manufacturer,
+        series=series,
+        mpn=mpn, num_rows=number_of_rows, pins_per_row=pins_per_row,
+        pitch=pitch, orientation=orientation_str)
+
+    kicad_mod = Footprint(footprint_name)
+    kicad_mod.setDescription("Molex {:s}, {:s} (compatible alternatives: {:s}), {:d} Pins per row ({:s}), generated with kicad-footprint-generator".format(series_long, mpn, ', '.join(alt_mpn), pins_per_row, variant_params[variant]['datasheet']))
+    kicad_mod.setTags(configuration['keyword_fp_string'].format(series=series,
+        orientation=orientation_str, man=manufacturer,
+        entry=configuration['entry_direction'][orientation]))
+
+    #kicad_mod.setAttribute('smd')
+
+    ########################## Dimensions ##############################
+    B = (pins_per_row-1)*pitch
+    A = B + 6.65
+    C = B + variant_params[variant]['C_minus_B']
+
+    pad_row_1_y = 0
+    pad_row_2_y = pad_row_1_y + pitch
+    pad1_x = 0
+
+    peg1_x = (B-C)/2
+    peg2_x = (B+C)/2
+    peg_y = pad_row_2_y + 0.94
+
+    tab_w = 1.4
+    tab_l = 1.4
+
+    body_edge={
+        'left':  (B-A)/2,
+        'right': (A+B)/2,
+        'bottom': 3.43 + pitch/2,
+        }
+    body_edge['top'] = body_edge['bottom'] - (7.37-0.5)
```

Comment:
It might be better to set 
```
body_edge['top'] = -1.47+0.5
body_edge['bottom'] = body_edge['top'] + (7.37-0.5)
```

Reply:
Adopted it to your suggestion. This makes it much more readable. I was also wondering where you got the 0.5 mm from and basically took it from your other footprints.